### PR TITLE
James 1772

### PR DESCRIPTION
**Pull request by rouazana**
Make some tests a little faster

### CODE REVIEW

**Comment by rouazana** [comment]
The 2 last commits are separated, because I'm not sure this little refactoring is interesting.

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraCluster.java [hunk 1]
```diff
@@ -80,7 +80,7 @@ public void clearAllTables() {
     private Optional<Session> tryInitializeSession() {
         try {
             Cluster clusterWithInitializedKeyspace = ClusterWithKeyspaceCreatedFactory
-                .clusterWithInitializedKeyspace(getCluster(), KEYSPACE_NAME, REPLICATION_FACTOR);
+                .clusterWithInitializedKeyspaceWithoutDurableWrites(getCluster(), KEYSPACE_NAME, REPLICATION_FACTOR);
```
Comment (chibenwa) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraCluster.java]: Really, really cool ! 
However, I wonder if it gets apply with the EmbbeddedCassandra during jmap-integration-testing. Could you check ?

File: server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/CassandraSessionModule.java [hunk 2]
```diff
@@ -76,12 +76,14 @@ Cluster provideCluster(FileSystem fileSystem, AsyncRetryExecutor executor) throw
...
-                                configuration.getInt("cassandra.port")),
-                        configuration.getString("cassandra.keyspace"),
-                        configuration.getInt("cassandra.replication.factor")))
+                .getWithRetry(ctx -> ClusterWithKeyspaceCreatedFactory
+                        .config(
+                            ClusterFactory.createClusterForSingleServerWithoutPassWord(
+                                    configuration.getString("cassandra.ip"),
+                                    configuration.getInt("cassandra.port")),
+                            configuration.getString("cassandra.keyspace"))
+                        .replicationFactor(configuration.getInt("cassandra.replication.factor"))
```
Comment (chibenwa) [inline server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/CassandraSessionModule.java]: I could be cool to expose durable writes here. You could then overright the default "false" value in the test in order to take advantage of this speed up for jmap-integration testing.
What do you think ?

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraCluster.java (hunk 1 above)
Comment (rouazana) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraCluster.java]: Of course I did, that's all the point of this commit :)

**Comment by chibenwa** [comment]
:+1:

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/ClusterWithKeyspaceCreatedFactory.java [hunk 3]
```diff
@@ -26,9 +26,17 @@
 
     private final static int DEFAULT_REPLICATION_FACTOR = 1;
 
+    public static Cluster clusterWithInitializedKeyspaceWithoutDurableWrites(Cluster cluster, String keyspace, int replicationFactor) {
```
Comment (mbaechler) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/ClusterWithKeyspaceCreatedFactory.java]: Maybe it's time to use a builder to expose all features ?

Comment (mbaechler) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/ClusterWithKeyspaceCreatedFactory.java]: ok, it's next commit (:

Comment (rouazana) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/ClusterWithKeyspaceCreatedFactory.java]: Yep, I can merge the 2 commits if you want.

**Comment by rouazana** [comment]
test this please

**Comment by mbaechler** [comment]
:+1:

**Comment by aduprat** [comment]
:+1:

**Pull request merged**